droidlet/lowlevel/locobot/remote/slam_pkg/slam.py

```diff
@@ -304,6 +359,34 @@ def take_step(self, step_size):
             return False
         return None
 
+    def save_rgb_depth_seg(self):
+        rgb, depth, seg = self.robot.camera.get_rgb_depth_segm()
```

Comment:
you can simply skip executing this function if `self.skp % 10 == 0` right?

So, add the logic to the top of this function:

```
if self.skp % 10 != 0:
    return
```

This way, you don't waste perf / cycles on pulling the rgb/depth on frames that you can simply skip

Reply:
similarly, you can do another perf optimization here, where you first pull `pos`, and then only if `pos != self.last_pos` do you pull the rgb / depth / seg

Reply:
did both.